From 95e60867b3172eb64e3f720a3457923f70ea820c Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 1 Jul 2019 11:24:13 +0800
Subject: [PATCH] pyanaconda: payload: Fix xserver detection

When installing an image that has graphical interface,
/etc/systemd/system/default.target will point to multi-user.target
instead of graphical.target.

To fix this, use the right rpm query to detect the presence of xserver.

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Update context for anaconda 32.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 pyanaconda/modules/services/installation.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyanaconda/modules/services/installation.py b/pyanaconda/modules/services/installation.py
index ead5b02fa2..9dc85b7982 100644
--- a/pyanaconda/modules/services/installation.py
+++ b/pyanaconda/modules/services/installation.py
@@ -218,8 +218,8 @@ class ConfigureSystemdDefaultTargetTask(Task):
         else:
             ts = rpm.TransactionSet(conf.target.system_root)
 
-            if ts.dbMatch("provides", 'service(graphical-login)').count():
-                log.debug("A package with provides == service(graphical-login) is installed, "
+            if ts.dbMatch("provides", 'xserver-xorg').count():
+                log.debug("A package with provides == xserver-xorg is installed, "
                           "using graphical.target.")
                 self._default_target = GRAPHICAL_TARGET
 
-- 
2.7.4

